src/services/queue-service.js

```diff
@@ -1,10 +1,40 @@
 import { UserNetworkInstance } from './network';
+import { setKeyWithExpiryToLocalStore } from '../utilities/local-storage';
+import { log } from "../utilities/log";
 
+//const DEFAULT_POLLING_DELAY_IN_SECONDS = 600;
+const DEFAULT_POLLING_DELAY_IN_SECONDS = 10;
+var currentPollingDelayInSeconds = DEFAULT_POLLING_DELAY_IN_SECONDS;
+var checkInProgress = false;
+
+export const userQueueNextPullCheckLocalStoreName = "gist.web.userQueueNextPullCheck";
 export async function getUserQueue() {
+  var response;
   try {
-    var response = await UserNetworkInstance().post(`/api/v1/users`, {});
-    return response;
+    if (!checkInProgress) {
+      checkInProgress = true;
+      response = await UserNetworkInstance().post(`/api/v1/users`, {});
+    }
   } catch (error) {
-    return error.response;
+    if (error.response) {
+      response = error.response;
+    } else {
+      log(`Error getting user queue: ${error}`);
+    }
+  } finally {
+    checkInProgress = false;
+    scheduleNextQueuePull(response);
+    return response;
+  }
+}
+
+function scheduleNextQueuePull(response) {
+  if (response && response.headers) {
+    var pollingInterval = response.headers['x-gist-queue-polling-interval'];
+    if (pollingInterval && pollingInterval > 0) {
```

Comment:
May be we could be more cautious here, where interval response might be too short or too long.

Reply:
We're determining this server side so I wouldn't be worried. I'd prefer having the flexibility to set it to whatever we need for a particular org.